lang/test/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransformTest.kt

```diff
@@ -7402,6 +7407,99 @@ class StaticTypeInferenceVisitorTransformTest : VisitorTransformTestBase() {
 
             return incompatibleTypeForIndex + validIndexType
         }
+        private fun createAggFunctionValidTests(
+            functionName: String,
+            inputTypes: StaticType,
+            expectedType: StaticType,
+            isTopLevel: Boolean
+        ): TestCase =
+            // testing toplevel aggregated function
+            // testing sql function(t)
+            // global environment here is a bag type
+            if (isTopLevel) {
+                TestCase(
+                    name = "top level $functionName(t)",
+                    originalSql = "$functionName(t)",
+                    globals = mapOf("t" to BagType(inputTypes)),
+                    handler = expectQueryOutputType(expectedType)
+                )
+            }
+            // testing aggregate function with select
+            // testing sql: SELECT function(t.a) as result FROM t
+            // global variable here is a bag of struct
+            else {
+                TestCase(
+                    name = "SELECT $functionName(t.a) as result FROM t",
+                    originalSql = "SELECT $functionName(t.a) as result FROM t",
+                    globals = mapOf("t" to BagType(unionOf(inputTypes.allTypes.map { StructType(mapOf("a" to it)) }.toSet()))),
+                    handler = expectQueryOutputType(BagType(StructType(mapOf("result" to expectedType), contentClosed = true)))
+                )
+            }
+
+        @JvmStatic
+        @Suppress("unused")
+        fun parametersForAggFunctionTests() =
+            // valid tests
+            listOf(
+                // count
```

Comment:
For COUNT, MIN, MAX, could have some tests with unknown values (e.g. `NULL`, `MISSING`, null/missing with a non-unknown type)